### PR TITLE
feature/139-tests-for-partner

### PR DESCRIPTION
**Pull request by MRomalde**
Created test for PartnerViewSet

You can test them with the command python manage.py test

### CODE REVIEW

**Review by pedlopruz — DISMISSED** [review]
The tests are working properly, good work.

**Review by manortbla — COMMENTED** [review]
The CRUD tests pass correctly  but it would be nice to also check that it also fails where it should, for example, create a test that fails if you exceed the number of characters in the address.

File: nexong/api/Event/tests.py [hunk 1]
```diff
@@ -776,3 +776,121 @@ def test_update_lesson_event_by_volunteer(self):
...
+            education_center=self.education_center2,
+            is_morning_student=True,
+            activities_during_exit="",
+            status="PENDIENTE",
+            current_education_year="TRES AÑOS",
+            education_center_tutor="Don Pablo Motos",
+            nationality="España",
+            birthdate="2015-04-21",
+            family=self.family2,
+        )
```
Comment (auroranavas) [inline nexong/api/Event/tests.py:831]: Using unrealistic data is one of the failure conditions for this subject... Please, change this.

File: nexong/api/Meeting/tests.py [hunk 2]
```diff
@@ -62,3 +62,74 @@ def test_delete_donations_by_admin(self):
...
+        )
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+
+    def test_update_meeting_by_partner(self):
+        attendees_ids = [self.partner3.id, self.partner4.id]
+        response = self.client.put(
+            f"/api/meeting/{self.meeting2.id}/",
+            data={
+                "name": "PruebaPapi",
+                "description": "Prueba Chulo",
```
Comment (auroranavas) [inline nexong/api/Meeting/tests.py:104]: 😑

**Review by auroranavas — CHANGES_REQUESTED** [review]
I get 4 failures. Even from tests that are already in develop. Check the comments I left in the code as well.

image

image

File: nexong/api/Event/tests.py (hunk 1 above)
Comment (MRomalde) [reply nexong/api/Event/tests.py:831]: Done the changes!

File: nexong/api/Meeting/tests.py (hunk 2 above)
Comment (MRomalde) [reply nexong/api/Meeting/tests.py:104]: Done changes!

**Comment by MRomalde** [comment]
I have made the changes for the tests. thanks you for the feedback!!

File: nexong/api/Authentication/tests.py [hunk 3]
```diff
@@ -45,14 +45,14 @@ def setUp(self):
...
-        self.user2 = User.objects.create(
+        self.volunteer3 = Volunteer.objects.create(**self.volunteer_data)
+        self.user3 = User.objects.create(
             username="testuser",
             email="example@gmail.com",
-            role=VOLUNTEER,
-            volunteer=self.volunteer2,
+            role=ADMIN,
+            volunteer=self.volunteer3,
         )
```
Comment (auroranavas) [inline nexong/api/Authentication/tests.py:54]: A volunteer can't be an ADMIN... The role should be VOLUNTEER as you had it before. This is why now the tests pass. You should try to see what the problem is, not change the code as you see fit.

**Review by auroranavas — CHANGES_REQUESTED** [review]
You should fix the errors, not change the assertEquals or the business logic so the tests pass. If we were to do this with all the tests, then, what purpose do they serve?

File: nexong/api/Authentication/tests.py (hunk 3 above)
Comment (MRomalde) [reply nexong/api/Authentication/tests.py:54]: the problem is something with the permissions. Not with the tests. I changed it to probe that the tests are OK

**Comment by MRomalde** [comment]
I mean, i changed it to probe that the tests are OK, there is something bad with the permissions in volunteer section.

**Comment by MRomalde** [comment]
I thought I writted the comment about, to probe it if there is any error now because i put the rol to ADMIN, but i did not writted it. Totally my bad.

**Comment by MRomalde** [comment]
This is the comment i did not write:
Please check if you have some failure now, I changed the role to ADMIN to probe the tests are ok, but we have some kind of error in volunteer permissions that i cant reach. thanks

**Comment by MRomalde** [comment]
Now its ok! The failure was that we were trying to update, retrieve and delete a volunteer that is not you, so you cant do that. Thats why we had a 403 error Forbidden!

**Review by auroranavas — APPROVED** [review]
Thank you for the changes. Good job!

**Pull request merged**